File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/repository/RimapPOIListResource.java
package de.deutschebahn.bahnhoflive.repository;

import java.util.List;

import de.deutschebahn.bahnhoflive.backend.rimap.model.RimapPOI;

public class RimapPOIListResource extends RemoteResource<List<RimapPOI>> {

    private Station stationId;

    @Override
    public boolean isLoadingPreconditionsMet() {
        return stationId != null;
    }

    @Override
    protected void onStartLoading(boolean force) {
        baseApplication.getRepositories().getMapRepository().queryPois(stationId, new Listener(), !force);
    }

    public void initialize(Station stationId) {
        this.stationId = stationId;
    }

    public void load() {
        loadData(false);
    }
}

File: modules/core/src/main/java/de/deutschebahn/bahnhoflive/repository/RimapPOIListResource.kt
/*
 * SPDX-FileCopyrightText: 2020 DB Station&Service AG <bahnhoflive-opensource@deutschebahn.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
package de.deutschebahn.bahnhoflive.repository

import de.deutschebahn.bahnhoflive.backend.rimap.model.RimapPOI

class RimapPOIListResource : RemoteResource<List<RimapPOI>>() {
    private var stationId: Station? = null
    override val isLoadingPreconditionsMet: Boolean
        get() = stationId != null

    override fun onStartLoading(force: Boolean) {
        stationId?.let { stationId ->
            baseApplication.repositories.mapRepository.queryPois(stationId, Listener(), !force)
        }
    }

    fun initialize(stationId: Station?) {
        this.stationId = stationId
    }

    fun load() {
        loadData(false)
    }
}